### ISSUE TITLE
Problem with JacobiWeight?

### ISSUE THREAD

**Issue by MarcoFasondini** [JuliaApproximation/OrthogonalPolynomialsQuasi.jl#68]
α = -0.5
β = -0.5
w = JacobiWeight(α,β)
p = LanczosPolynomial(w)
x = axes(w,1)
b = 2
ϕw = JacobiWeight(α+1,β+1).*(b .- x)
pϕ = LanczosPolynomial(ϕw)

The following doesn't give a diagonal operator:
pϕ' * (ϕw .* pϕ)

p' * (w .* p)
gives
not implemented for OrthogonalPolynomialsQuasi.Jacobi{Float64}(0.0, 0.0) and OrthogonalPolynomialsQuasi.Jacobi{Float64}(-0.5, -0.5)

**Comment by dlfivefifty** [comment]
A simpler example with the same problem:
```julia
julia> P = Jacobi(0.5,0.5)
Jacobi{Float64}(0.5, 0.5)

julia> w = JacobiWeight(0.5,0.5)
JacobiWeight{Float64}(0.5, 0.5)

julia> P' * (w .* P)
ERROR: not implemented for Jacobi{Float64}(0.0, 0.0) and Jacobi{Float64}(0.5, 0.5)
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] \(::Jacobi{Float64}, ::Jacobi{Float64}) at /Users/sheehanolver/Projects/OrthogonalPolynomialsQuasi.jl/src/jacobi.jl:227
 [3] \ at /Users/sheehanolver/Projects/OrthogonalPolynomialsQuasi.jl/src/jacobi.jl:205 [inlined]
 [4] legendre_massmatrix(::QuasiAdjoint{Float64,Jacobi{Float64}}, ::BroadcastQuasiArray{Float64,2,typeof(*),Tuple{JacobiWeight{Float64},Jacobi{Float64}}}) at /Users/sheehanolver/Projects/OrthogonalPolynomialsQuasi.jl/src/jacobi.jl:144
 [5] macro expansion at /Users/sheehanolver/Projects/OrthogonalPolynomialsQuasi.jl/src/jacobi.jl:160 [inlined]
 [6] mul at /Users/sheehanolver/Projects/ContinuumArrays.jl/src/operators.jl:33 [inlined]
 [7] *(::QuasiAdjoint{Float64,Jacobi{Float64}}, ::BroadcastQuasiArray{Float64,2,typeof(*),Tuple{JacobiWeight{Float64},Jacobi{Float64}}}) at /Users/sheehanolver/.julia/packages/QuasiArrays/9YqBZ/src/matmul.jl:32
 [8] top-level scope at REPL[1090]:1
 [9] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1088
```

Will try to fix this.

PS The "Not implemented" should actually work as its just the Jacobi-Jacobi conversion, but haven't got around to adding it. The formula is in Townsend–Webb–Olver

**Comment by dlfivefifty** [comment]
So by hook and by crook I got this fixed. 

At the moment it's special cased for `Weight .* AffineQuasiVector`, that is, it won't work for something like `JacobiWeight(α+1,β+1).*(b .- x).^2`.
